### ISSUE TITLE
Different MAP Elites Bins for Zelda

### ISSUE THREAD

**Issue by schrum2** [schrum2/MM-NEAT#441]
Encourage diversity within the dungeon itself. One way of doing this would be to have a binning dimension for the number of unique rooms in the dungeon. Can calculate by adding all dungeon rooms to a Set data structure (which disallows duplicates) and then checking the size of the Set afterward. Might require redefinition of equality operation to ignore doors.

**Comment by schrum2** [comment]
This should not be your first task, but you should get around to it eventually.

The trick here will be to define a new way of binning the dungeons while still maintaining the option of doing it the old way (backwards compatibility). Part of this will involve renaming ZeldaMAPElitesBinLabels to something more specific, and creating a new class that describes the alternative binning approach. However, this alone is not enough, because ZeldaDungeonTask is currently hard-coded to bin things in a way that matches ZeldaMAPElitesBinLabels.

Basically, the code that sets the binIndex (look under if(MMNEAT.ea instanceof MAPElites)) needs to have different cases for different binning schemes. It would be nice if this code could be moved elsewhere, but this will require some discussion in order to do correctly.

**Comment by schrum2** [comment]
You should start this issue soon. Some discussion with me will be required to make sense of changing the binning scheme. However, even before you do that, there is something you should be able to figure out on your own. In the evaluate method of ZeldaDungeonTask, define a variable that calculates/tracks the number of distinct rooms in the dungeon using the Set data structure as described above. You won't actually be doing anything with this value yet, but you need a way of calculating it before you can move to using it in a binning scheme

**Comment by schrum2** [comment]
Transferring this issue to @cappsb since he already has experience defining fitness functions in the ZeldaDungeonTask class

**Comment by cappsb** [comment]
![image](https://user-images.githubusercontent.com/63438571/82097866-611b6680-96c9-11ea-83b1-77d2a6e43721.png)

Says 25 distinct rooms but there are 3 or more repeated rooms.

**Comment by schrum2** [comment]
Related to this: Create a directory `batch\Experiments-2020-MoreCPPN2GAN` and copy over the files `MAPElitesZeldaCPPNtoGAN-RoomsTraversed.bat` and `MAPElitesZeldaGAN-RoomsTraversed.bat` from `batch\Experiments-2020-GECCO-CPPN2GAN`. However, you will then need to rename these files and also edit them (with a text editor) in the following ways:
1. Change command line parameters to use your new binning scheme, along with the new fitness calculations
2.  For the `saveTo` and `log` parameters, change the `MAPElites` part to `NewMAPElites`

Let me know when this is done and I'll test it out
